packages/mdc-chips/chip/index.js

```diff
@@ -65,8 +65,44 @@ class MDCChip extends MDCComponent {
       addClass: (className) => this.root_.classList.add(className),
       removeClass: (className) => this.root_.classList.remove(className),
       hasClass: (className) => this.root_.classList.contains(className),
+      addClassToLeadingIcon: (className) => {
+        const leadingIconEl = this.root_.querySelector(strings.LEADING_ICON_SELECTOR);
+        if (leadingIconEl) {
+          leadingIconEl.classList.add(className);
+        }
+      },
+      removeClassFromLeadingIcon: (className) => {
+        const leadingIconEl = this.root_.querySelector(strings.LEADING_ICON_SELECTOR);
+        if (leadingIconEl) {
+          leadingIconEl.classList.remove(className);
+        }
+      },
       registerInteractionHandler: (evtType, handler) => this.root_.addEventListener(evtType, handler),
       deregisterInteractionHandler: (evtType, handler) => this.root_.removeEventListener(evtType, handler),
+      registerLeadingIconEventHandler: (evtType, handler) => {
+        const leadingIconEl = this.root_.querySelector(strings.LEADING_ICON_SELECTOR);
```

Comment:
Can we store the leading icon and checkmark elements in instance properties so we don't have to do the DOM lookup every time these are called? e.g. what we did in text-field with `this.input_`